### PR TITLE
ICU-22742 Fix handling of XA,XB,XC for addLikelySubtags

### PR DESCRIPTION
**Pull request by FrankYFTang**
##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22742
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [X] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/common/loclikelysubtags.cpp [hunk 1]
```diff
@@ -565,24 +565,16 @@ LSR LikelySubtags::makeMaximizedLsr(const char *language, const char *script, co
     // They should match only themselves,
     // not other locales with what looks like the same language and script subtags.
     char c1;
-    if (region[0] == 'X' && (c1 = region[1]) != 0 && region[2] == 0) {
+    if (region[0] == 'X' && (c1 = region[1]) != 0 && region[2] == 0 &&
```
Comment (mihnita) [inline icu4c/source/common/loclikelysubtags.cpp:568]: This does not seem to fix the `PS*` variants.

Just wrapping both `if`s seems to do the job:
```java
if (!returnInputIfUnmatch) {
    if (region[0] == 'X' && (c1 = region[1]) != 0 && region[2] == 0) {
        ...
    }
    if (variant[0] == 'P' && variant[1] == 'S') {
        ...
    }
}
```

Comment (mihnita) [reply icu4c/source/common/loclikelysubtags.cpp:568]: Same applies to icu4c, of course

**Comment by FrankYFTang** [comment]
PTAL

**Review by mihnita — APPROVED** [review]
Thank you very much!
Mihai

**Pull request merged**